hedera-mirror-rest/utils.js

```
@@ -983,11 +997,63 @@ const loadPgRange = () => {
   pgRange.install(pg);
 };
 
+//Convert maxTimestampRange from milliseconds to nanoseconds
+const maxTimestampRange = math.multiply(config.maxTimestampRange, 1e6);
+/**
+ * Checks that the timestamp filters contains either a valid range (greater than and less than filters that do not span beyond
+ * a configured limit), or at least one equals operator.
+ *
+ * @param timestamps a timestamp or array of timestamps directly from the req
+ * @returns boolean whether or not the timestamps created a range (within the limits) or contain an equals operator
+ */
+const checkTimestampRange = (timestamps) => {
+  let latest = undefined;
+  let earliest = undefined;
+
+  //no timestamp params
+  if (!timestamps) {
+    return false;
+  }
+
+  const timestampsArray = Array.isArray(timestamps) ? timestamps : [timestamps];
+
+  for (const val of timestampsArray) {
+    const filter = buildComparatorFilter(constants.filterKeys.TIMESTAMP, val);
+    if (filter.operator === 'eq') {
+      //An equals operator removes the need for a range
+      return true;
+    } else if (filter.operator === `gt` || filter.operator === 'gte') {
+      if (earliest !== undefined) {
+        //Multiple greater than operators detected, not permitted
```

Is there a way to return these comments as validation error messages on the response? Perhaps instead of returning boolean we can throw an error with the message. We can catch the error higher up to support returning multiple validation errors. Otherwise the user will have a difficult time figuring out what's wrong with their input.

---

Rewrote this to throw clear error instead of returning booleans